src/sources/splunk_hec/mod.rs

```diff
@@ -161,8 +171,75 @@ impl SourceConfig for SplunkConfig {
         }))
     }
 
-    fn outputs(&self, _global_log_namespace: LogNamespace) -> Vec<Output> {
-        vec![Output::default(DataType::Log)]
+    fn outputs(&self, global_log_namespace: LogNamespace) -> Vec<Output> {
+        let log_namespace = global_log_namespace.merge(self.log_namespace);
+
+        let schema_definition = match log_namespace {
```

Comment:
Looking at this (first time), it sorta seems like this is either going to be either the Json decoder or the Bytes decoder - is there a way we can reuse that?

Reply:
It's all a json object. This object has a field `event` that contains the message. That message can be a String, or an object that contains a field `line` that contains an Object or a String.

https://github.com/vectordotdev/vector/blob/stephen/splunk_log_namespace/src/sources/splunk_hec/mod.rs#L664-L688

Reply:
🤔 is there a reason why we're not just using the JSON deserializer then?

Reply:
I guess because it only extracts specific fields in order to build up the event.

Reply:
Hmmm, ok - I'll defer to your judgement/knowledge of the source here. Just looking for opportunities for code-reuse over snowflakes 👍